src/components/modals/authentification/ModalAuth.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * MODAL inscription / connexion
+ * LOGIQUE D'AFFICHAGE ENTRE CONNEXION - INSCRIPTION - MOT DE PASSE OUBLIE
+ */
+
+import { createPortal } from 'react-dom';
+import { useState } from 'react';
+import { X } from 'lucide-react';
+import ModalAuthInitView from './ModalAuthInitView';
+import ModalAuthConnexionView from './ModalAuthConnexionView';
+import ModalAuthInscription from './ModalAuthInscription';
+import ModalPasswordForget from './ModalPasswordForget';
+import useFirebaseAuth from '@/hooks/useFirebaseAuth';
+import ButtonDisconnect from './ButtonDisconnect';
+
+const ModalAuth = () => {
+  const [contenuModal, setContenuModal] = useState<
+    'init' | 'connexion' | 'inscription' | 'password-forget'
+  >('init');
+
+  const { authUser } = useFirebaseAuth();
+
+  return createPortal(
+    <>
+      {/* Fond gris */}
+      <div className="fixed h-screen top-0 left-0 bottom-0 right-0 bg-vintedTextGrisClair opacity-10"></div>
+
+      {/* Modal */}
+      <div
+        className="absolute w-full top-12 sm:w-[375px] sm:top-1/2 sm:left-1/2 sm:-translate-x-1/2 sm:-translate-y-1/2  bg-vintedBackground flex flex-col items-center justify-start rounded"
+        aria-modal="true"
+        role="dialog"
```

Comment:
Amélioration repérée avec Sonarlint : Use <dialog> instead of the "dialog" role to ensure accessibility across all devices.sonarlint(typescript:S6819)

Reply:
pas compris

Reply:
Il manque un bout du message : Use **<dialog>** instead of the "dialog" role to ensure accessibility across all devices.sonarlint(typescript:S6819)

Reply:
corrigé, j'ai supprimé dialog